### PR TITLE
Setup script for Node.js webhooks example

### PR DESCRIPTION
**Pull request by aw-asana**
- Introduces **createWebhook.js**, a setup script for establishing webhooks and validating the stored `X-Hook-Secret` against the one received from Asana's `201` response during webhook creation. This validation checks that the secrets match (i.e., enhance security by allowing only legitimate sources to trigger events).
- Updates the README to describe these changes in installation/usage

### CODE REVIEW

File: javascript/webhooks-nodejs/README.md [hunk 1]
```diff
@@ -28,31 +28,47 @@ Ngrok is used to create a publicly accessible "tunnel" (i.e., URL) to a port on
 
 ## Usage
 
-1. Create a file named **.env** in the root directory of this app. In the file, include the following line:
+After ngrok is successfully running, the next steps are to run the server (**index.js**), then establish a webhook using the provided webhook setup script (**createWebhook.js**).
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/README.md:31]: **Nit:** maybe we should tell developers how to start the ngrok server. I think [by default we set it to port](https://github.com/Asana/devrel-examples/blob/aw-asana-webhooks-setup/javascript/webhooks-nodejs/index.js#L67-L69) `8080` any ways so we might as well tell them to run `ngrok http 8080`

File: javascript/webhooks-nodejs/README.md [hunk 2]
```diff
@@ -28,31 +28,47 @@ Ngrok is used to create a publicly accessible "tunnel" (i.e., URL) to a port on
...
 ```
 npm run dev
 ```
 
-3. Create a webhook by calling `POST /webhooks` and specifying given resource (e.g., a task). Feel free to use the [Postman Collection](https://developers.asana.com/docs/using-postman) or the [POST /webhooks](https://developers.asana.com/reference/createwebhook) page in the API Explorer to make your request.
+3. Edit the **createWebhook.js** setup script to include details for your [POST /webhooks](https://developers.asana.com/reference/createwebhook) request:
+
+```js
+// TODO: Replace these values with your target URI, object ID, filter, and resource type
+const targetUri = "https://your.url.here/receiveWebhook"; // The webhook server's public endpoint for receiving webhooks
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/README.md:56]: Nit: `your.url.here` can we use `<>` convention here it makes it more clear and it's more common in our docs with the client libraries.

TLDR; `your.url.here` -> `<YOUR_URL_HERE>`

File: javascript/webhooks-nodejs/createWebhook.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  } catch (error) {
+    console.error(
+      "Error creating webhook:",
+      error.response ? error.response.data : error.message
+    );
+  }
+}
+
+// TODO: Replace these values with your target URI, object ID, filter, and resource type
+const targetUri = "https://your.url.here/receiveWebhook"; // The webhook server's public endpoint for receiving webhooks
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/createWebhook.js:97]: Same as comment above:

Nit: `your.url.here` can we use `<>` convention here it makes it more clear and it's more common in our docs with the client libraries.

TLDR; `your.url.here` -> `<YOUR_URL_HERE>`

File: javascript/webhooks-nodejs/createWebhook.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        `X-Hook-Secrets do not match! Potential security issue. Deleting webhook with ID ${response.data.data.gid}`
+      );
+      // Delete the webhook if the secrets don't match
+      await deleteWebhook(response.data.data.gid, personalAccessToken);
+      return;
+    }
+
+    updateXHookSecret(xHookSecret);
+    console.log(
+      `Webhook created successfully! The X-Hook-Secret from Asana's 201 response was: ${xHookSecret}`
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/createWebhook.js:86]: Nit: Can we also log the webhook GID. I feel like that would be useful for developers if they wan to call [Get a webhook](https://app.asana.com/0/1199684513975168/1205105675089975)

File: javascript/webhooks-nodejs/createWebhook.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    console.error(
+      "Error creating webhook:",
+      error.response ? error.response.data : error.message
+    );
+  }
+}
+
+// TODO: Replace these values with your target URI, object ID, filter, and resource type
+const targetUri = "https://your.url.here/receiveWebhook"; // The webhook server's public endpoint for receiving webhooks
+const objectId = "TASK_ID_HERE"; // The Asana object ID you want to track
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/createWebhook.js:98]: `// The Asana object ID you want to track` and `TASK_ID_HERE` is conflicting maybe add something like `(e.g., task gid)`

File: javascript/webhooks-nodejs/createWebhook.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        },
+      }
+    );
+
+    const xHookSecret = response.data["X-Hook-Secret"];
+    const storedSecret = getXHookSecret();
+
+    // Compare the secrets
+    // Read more about validation during the webhook handshake: https://developers.asana.com/docs/webhooks-guide#the-webhook-handshake
+    if (xHookSecret !== storedSecret) {
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/createWebhook.js:75]: Hmm not sure if I can trigger this case. I might need you to show me how to hit this case. I tried changing my `.env` file's `X-Hook-Secret` before running and I don't encounter this case. Might be my terminal or something

File: javascript/webhooks-nodejs/README.md (hunk 1 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/README.md:37]: Sounds good to me 👍

File: javascript/webhooks-nodejs/README.md (hunk 2 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/README.md:62]: Fixed!

File: javascript/webhooks-nodejs/createWebhook.js (hunk 3 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/createWebhook.js]: Fixed!

File: javascript/webhooks-nodejs/createWebhook.js (hunk 5 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/createWebhook.js]: Fixed!

File: javascript/webhooks-nodejs/createWebhook.js (hunk 4 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/createWebhook.js]: Fixed!

File: javascript/webhooks-nodejs/createWebhook.js (hunk 6 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/createWebhook.js:75]: Hmm I'm unable to re-produce the use case above. I created a random helper function to programmatically set `X_HOOK_SECRET` in the .env file to `abc`:

```js
function setXHookSecretToAbc() {
  const newSecret = "abc";
  let envContent = fs.readFileSync(envFilePath, "utf8");
  envContent = envContent.replace(
    /X_HOOK_SECRET=.*/,
    `X_HOOK_SECRET=${newSecret}`
  );
  fs.writeFileSync(envFilePath, envContent);
  console.log(`The X-Hook-Secret in .env has been updated to: ${newSecret}.`);
}
```

Then I called it just before I compared the stored secret vs the one from Asana's `201`:

```js
const xHookSecret = response.data["X-Hook-Secret"];

console.log(`The original X_HOOK_SECRET from .env before the re-write is ${getXHookSecret()}`)
setXHookSecretToAbc();
console.log(
  `After the re-writen, the X_HOOK_SECRET from .env re-write is now ${getXHookSecret()}`
);

const storedSecret = getXHookSecret();
```

The console output was:

```bash
The original X_HOOK_SECRET from .env before the re-write is 11111111111111111111111111111111
The X-Hook-Secret in .env has been updated to: abc.
After the re-writen, the X_HOOK_SECRET from .env re-write is now abc
X-Hook-Secrets do not match! Potential security issue. Deleting webhook with GID 2222222222222222.
Webhook 2222222222222222 deleted due to mismatch.
```

File: javascript/webhooks-nodejs/README.md (hunk 2 above)
Comment (jv-asana) [reply javascript/webhooks-nodejs/README.md:62]: It looks like you fixed this in the `createWebhook.js` file but not the `README.md`

File: javascript/webhooks-nodejs/README.md [hunk 7]
```diff
@@ -26,33 +26,55 @@ https://ngrok.com/docs/getting-started/
...
 ```
 
-3. Create a webhook by calling `POST /webhooks` and specifying given resource (e.g., a task). Feel free to use the [Postman Collection](https://developers.asana.com/docs/using-postman) or the [POST /webhooks](https://developers.asana.com/reference/createwebhook) page in the API Explorer to make your request.
+3. Edit the **createWebhook.js** setup script to include details for your [POST /webhooks](https://developers.asana.com/reference/createwebhook) request:
 
-> **Note:** In your request, make sure to set the [`target`](https://developers.asana.com/docs/webhook) parameter to your public ngrok domain instead of the example `localhost` domain, `http://localhost:8000/receiveWebhook`. This means you must replace `localhost:8000` with your unique ngrok domain. The `target` should be your ngrok server's "Forwarding" domain followed by `/receiveWebhook`. 
+```js
+// TODO: Replace these values with your target URI, object ID, filter, and resource type
+const targetUri = "https://your.url.here/receiveWebhook"; // The webhook server's public endpoint for receiving webhooks
+const objectId = "TASK_ID_HERE"; // The Asana object ID you want to track
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/README.md:63]: You'll probably want to change this since you changed this in the `createWebhook.js` also to make it consistent let's add the`<` and `>` around the placeholder value `"TASK_ID_HERE"` -> `<OBJECT_ID_HERE>`

File: javascript/webhooks-nodejs/createWebhook.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        headers: {
+          Authorization: `Bearer ${personalAccessToken}`,
+        },
+      }
+    );
+
+    const xHookSecret = response.data["X-Hook-Secret"];
+    const storedSecret = getXHookSecret();
+
+    // Compare the secrets. This check is not solely for detecting mutations in the stored secret; rather, it ensures that the original secret was valid and corresponds to what Asana expects
```
Comment (jv-asana) [inline javascript/webhooks-nodejs/createWebhook.js:73]: **Nit:** Can you add new lines to this comment so we don't have to scroll horizontally to view
scroll_horizontally

File: javascript/webhooks-nodejs/README.md (hunk 7 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/README.md:63]: Fixed! I originally left it out so that developers are not mislead that `<` and `>` belong in the request payload.

File: javascript/webhooks-nodejs/createWebhook.js (hunk 8 above)
Comment (aw-asana) [reply javascript/webhooks-nodejs/createWebhook.js]: Fixed!

File: javascript/webhooks-nodejs/createWebhook.js (hunk 6 above)
Comment (jv-asana) [reply javascript/webhooks-nodejs/createWebhook.js:76]: This case will be harder to test since it requires pretending to be Asana and sending back a 201 response with a fake `X-Hook-Secret`.  Adding a comment, here explaining why we do this check should help developers understand the purpose.

It looks like you've already added a comment so this should be good

**Review by jv-asana — APPROVED** [review]
LGTM
